### PR TITLE
Feature/sticker integration

### PR DESCRIPTION
**Pull request by nkhoshini**
Adds support for
- fetching sticker packs
- fetching an individual pack
- fetching stickers in a pack

REST API Source: https://developers.giphy.com/docs/

### CODE REVIEW

File: app/src/main/java/com/giphy/sdk/core/network/response/ListStickerPacksResponse.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
+package com.giphy.sdk.core.network.response;
+
+import com.giphy.sdk.core.models.Meta;
+import com.giphy.sdk.core.models.Pagination;
+import com.giphy.sdk.core.models.StickerPack;
+
+import java.util.List;
+
+/**
+ * Created by nkhoshini on 10/23/17.
```
Comment (bogdantmm92) [inline app/src/main/java/com/giphy/sdk/core/network/response/ListStickerPacksResponse.java]: nit: We might need attribution `This Source Code Form is subject to the terms of the Mozilla Public.....`

File: app/src/main/java/com/giphy/sdk/core/network/response/StickerPackResponse.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
+/*
+ * Created by Bogdan Tirca on 4/21/17.
```
Comment (bogdantmm92) [inline app/src/main/java/com/giphy/sdk/core/network/response/StickerPackResponse.java]: nit: Created by MR Nima :)

File: app/src/main/java/com/giphy/sdk/core/network/api/GPHApi.java [hunk 3]
```diff
@@ -159,5 +161,34 @@ public Future gifsByIds(@NonNull List<String> gifIds,
...
                                   @NonNull final CompletionHandler<ListTermSuggestionResponse> completionHandler);
+
+
+    /**
+     * Returns all sticker packs
+     * @param completionHandler
+     * @return
+     */
+    @NonNull
+    public Future getStickerPacks(@NonNull final CompletionHandler<ListStickerPacksResponse> completionHandler);
```
Comment (bogdantmm92) [inline app/src/main/java/com/giphy/sdk/core/network/api/GPHApi.java]: Does this endpoint also support limit, offset, rating, language?

Comment (bogdantmm92) [reply app/src/main/java/com/giphy/sdk/core/network/api/GPHApi.java]: Same for all 3 endpoints

File: app/src/main/java/com/giphy/sdk/core/models/StickerPack.java [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+  private String id;
+  @SerializedName("display_name")
+  private String displayName;
+  private String parent;
+  private String slug;
+  private String type;
+  @SerializedName("content_type")
+  private MediaType contentType;
+  @SerializedName("short_display_name")
+  private MediaType shortDisplayName;
```
Comment (bogdantmm92) [inline app/src/main/java/com/giphy/sdk/core/models/StickerPack.java]: I think this should be a String

File: app/src/main/java/com/giphy/sdk/core/models/StickerPack.java [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+  }
+
+  @Override
+  public int describeContents() {
+    return 0;
+  }
+
+  @Override
+  public void writeToParcel(Parcel dest, int flags) {
+    dest.writeString(this.id);
```
Comment (bogdantmm92) [inline app/src/main/java/com/giphy/sdk/core/models/StickerPack.java]: We need to write some tests for this. This is an example https://github.com/Giphy/giphy-android-sdk-core/blob/master/app/src/androidTest/java/com/giphy/sdk/core/SubcategoriesTest.java#L130

Comment (bogdantmm92) [reply app/src/main/java/com/giphy/sdk/core/models/StickerPack.java]: Also we need some tests that run the network requests and check if the most important fields are present, if pagination works...
https://github.com/Giphy/giphy-android-sdk-core/blob/master/app/src/androidTest/java/com/giphy/sdk/core/SubcategoriesTest.java#L98

File: app/src/main/java/com/giphy/sdk/core/network/api/GPHApi.java (hunk 3 above)
Comment (bogdantmm92) [reply app/src/main/java/com/giphy/sdk/core/network/api/GPHApi.java]: nit: would be ideal to stick to the naming convention. Don't have a strong opinion if the name should have `get` or not, but whatever convention we choose, it needs to be consistent across all methods, on both platforms.

File: app/src/main/java/com/giphy/sdk/core/network/api/Constants.java [hunk 6]
```diff
@@ -27,5 +27,8 @@
         public static final String GIF_BY_ID= "v1/gifs/%s";
         public static final String GIF_BY_IDS= "v1/gifs";
         public static final String TERM_SUGGESTIONS= "v1/queries/suggest/%s";
+        public static final String FETCH_STICKER_PACKS= "v1/stickers/packs";
```
Comment (bogdantmm92) [inline app/src/main/java/com/giphy/sdk/core/network/api/Constants.java]: nit: naming convention is without prefix

File: app/src/main/java/com/giphy/sdk/core/models/StickerPack.java (hunk 4 above)
Comment (nkhoshini) [reply app/src/main/java/com/giphy/sdk/core/models/StickerPack.java]: Yup! Good catch

File: app/src/main/java/com/giphy/sdk/core/models/StickerPack.java (hunk 5 above)
Comment (nkhoshini) [reply app/src/main/java/com/giphy/sdk/core/models/StickerPack.java]: Definitely, will add those

File: app/src/main/java/com/giphy/sdk/core/network/api/Constants.java (hunk 6 above)
Comment (nkhoshini) [reply app/src/main/java/com/giphy/sdk/core/network/api/Constants.java]: 👍

File: app/src/main/java/com/giphy/sdk/core/network/response/ListStickerPacksResponse.java (hunk 1 above)
Comment (nkhoshini) [reply app/src/main/java/com/giphy/sdk/core/network/response/ListStickerPacksResponse.java]: 💯

File: app/src/main/java/com/giphy/sdk/core/network/api/GPHApi.java (hunk 3 above)
Comment (nkhoshini) [reply app/src/main/java/com/giphy/sdk/core/network/api/GPHApi.java]: I do believe it supports paging, will look into this and update along with the naming consistencies

**Comment by nkhoshini** [comment]
@bogdantmm92 pushed over changes to address the issues you listed, let me know if there's other stuff I missed!

File: app/src/androidTest/java/com/giphy/sdk/core/StickerPacksTest.java [hunk 7]
```diff
@@ -0,0 +1,230 @@
...
+    }
+
+    /**
+     * Test getting a sticker pack and then its stickers with a limit and offset
+     *
+     * @throws Exception
+     */
+    @Test
+    public void testStickersByPackIdWithOffset() throws Exception {
+        final CountDownLatch lock = new CountDownLatch(2);
```
Comment (bogdantmm92) [inline app/src/androidTest/java/com/giphy/sdk/core/StickerPacksTest.java]: I think we need CountDownLatch(`1`) here.

File: app/src/androidTest/java/com/giphy/sdk/core/StickerPacksTest.java [hunk 8]
```diff
@@ -0,0 +1,230 @@
...
+    }
+
+    /**
+     * Test getting a sticker pack's child
+     *
+     * @throws Exception
+     */
+    @Test
+    public void testStickerChildren() throws Exception {
+        final CountDownLatch lock = new CountDownLatch(2);
```
Comment (bogdantmm92) [inline app/src/androidTest/java/com/giphy/sdk/core/StickerPacksTest.java]: And here

**Pull request merged**